gosub-bindings/src/lib.rs

```diff
@@ -0,0 +1,123 @@
+use std::ffi::CStr;
+use std::ffi::CString;
+use std::os::raw::c_char;
+use std::ptr;
+
+use gosub_engine::{
+    bytes::{CharIterator, Confidence, Encoding},
+    html5::parser::{
+        document::{Document, DocumentBuilder},
+        Html5Parser,
+    },
+    render_tree::{Node, NodeType, RenderTree, TreeIterator},
+};
+
+#[no_mangle]
+/// Initialize a render tree and return an owning pointer to it.
+/// If the HTML fails to parse, returns a NULL pointer.
+///
+/// # Safety
+/// Takes a read-only pointer owned from the C API representing the HTML source
+/// to build a render tree. DO NOT take ownership of this pointer in Rust or the
+/// universe might collapse.
+///
+/// Moves an owning pointer to the render_tree using Box::into_raw() to the C API.
+/// This pointer MUST be passed to gosub_render_tree_free() after usage for proper cleanup.
+pub unsafe extern "C" fn gosub_render_tree_init(html: *const c_char) -> *mut RenderTree {
```

Comment:
For now, this is okay, but i guess we want this most likely to be a URL and not the HTML document, so we can use the browser's dns system

Reply:
Yeah, I would want to move to the DNS system eventually, but was not available at the time I wrote this

Reply:
Are we guaranteed that the Swift Objective C side will never free this `*const c_char` pointer while we're in Rust code?